modules/layers/src/solid-polygon-layer/polygon.js

```diff
@@ -49,6 +50,30 @@ function isSimple(polygon) {
   return polygon.length >= 1 && polygon[0].length >= 2 && Number.isFinite(polygon[0][0]);
 }
 
+/**
+ * Computes winding order of a polygon and reverses points of the polygon in case of counter clockwise winding order.
+ * @param {Array} positions - array of numbers
+ * @param {Number} start - start index of the polygon in the positions array
+ * @param {Number} end - end index of the polygon in the positions array
+ * @param {Number} size - size of a position, 2 (xy) or 3 (xyz)
+ */
+function ensureWindingOrder(positions, start, end, size) {
+  const numPositions = (end - start) / size;
+  const points = [];
+  for (let i = 0; i < numPositions; i++) {
+    points[i] = [positions[i * size], positions[i * size + 1]];
+  }
+
+  const polygon = new Polygon(points);
+
+  if (polygon.getWindingDirection() < 0) {
```

Comment:
This is incorrect. The holes need to wind in the opposite direction from the outer bounds.

Reply:
But is it required for earcut or rendering?

Reply:
Not sure about earcut, but definitely rendering. You can test your solution locally with a polygon that contains holes.

Reply:
Btw that is a requirement from the GeoJSON spec and followed by all conforming formats e.g. MVT

Reply:
Correct, missed this